### DISCUSSION TITLE
Unable to create root user on setup

### DISCUSSION THREAD

**Discussion by fabiankaestner** [Budibase/budibase#3021] [Help]
I'm using the helm chart v0.2.0 to deploy budibase v0.9.163 on a single node k3s test cluster.
When I visit the URL for the budibase installation, I do not get the expected 'Create Root User' form, but instead a regular login form at `XXX/builder/auth/login`. Of course, I have not yet setup an account and am thus unable to login.

I am using the following values.yml:
```yaml
ingress:
  enabled: true
  nginx: false
  annotations:
    kubernetes.io/ingress.class: "traefik"
    cert-manager.io/cluster-issuer: letsencrypt-prod
    traefik.ingress.kubernetes.io/router.middlewares: default-redirect-https@kubernetescrd
  hosts:
    - host: XXX
      paths:
        - path: /
          pathType: Prefix
          backend:
            service:
              name: proxy-service
              port:
                number: 10000
  tls:
    - hosts:
        - XXX
      secretName: budibase-test-cert-tls
```

The couchdb pods report the following error:
```
[notice] 2021-10-15T14:50:57.949324Z couchdb@budibase-couchdb-0.budibase-couchdb.budibase.svc.cluster.local <0.362.0> -------- chttpd_auth_cache changes listener died because the _users database does not exist. Create the database to silence this notice.
[error] 2021-10-15T14:50:57.949603Z couchdb@budibase-couchdb-0.budibase-couchdb.budibase.svc.cluster.local emulator -------- Error in process <0.6760.0> on node 'couchdb@budibase-couchdb-0.budibase-couchdb.budibase.svc.cluster.local' with exit value:
{database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,399}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,374}]},{mem3_shards,load_shards_from_disk,2,[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,for_docid,3,[{file,"src/mem3_shards.erl"},{line,96}]},{fabric_doc_open,go,3,[{file,"src/fabric_doc_open.erl"},{line,39}]},{chttpd_auth_cache,ensure_auth_ddoc_exists,2,[{file,"src/chttpd_auth_cache.erl"},{line,198}]},{chttpd_auth_cache,listen_for_changes,1,[{file,"src/chttpd_auth_cache.erl"},{line,145}]}]}
2021-10-15T16:50:57.950013229+02:00
```

What am I doing wrong here?

**Comment by fabiankaestner** [comment]
So, I did some further debugging and I think I have discovered the issue. However, before debugging this further and possibly spreading wrong information here, I was wondering if one of the team would mind taking some time to chat about the helm chart as it is. @shogunpurple , if you have some time to chat, shoot me an email at mail@fabiankaestner.me.

**Reply by shogunpurple** [reply]
Hey @fabiankaestner, just seeing this.

I am unable to reproduce this locally on minikube, but it would be good to run some commands to get the logs of the app server. You can do this with:

`kubectl logs app-service -n budibase`

No problem, happy to chat about it further via email. I’ll send you an email now.

**Comment by PatrickHuetter** [comment]
I can confirm this issue with current latest 0.2.0 helm chart. It's not possible to create an admin user after fresh installation using a plain clean kubernetes cluster.

**Reply by shogunpurple** [reply]
Thanks @PatrickHuetter - @fabiankaestner has identified some missing configuration values in the helm chart that we will get resolved ASAP

**Reply by shogunpurple** [reply]
In the meantime, you should be able to update your `values.yaml` to have the following values:

`selfHosted: 1`

`services.objectStore.url: minio-service:9000`

if you update those, can you create your admin user?

**Reply by shogunpurple** [reply]
https://github.com/Budibase/budibase/pull/3040

**Reply by PatrickHuetter** [reply]
Now i get the admin creation screen but it's not possible to submit the form successfully.
<img width="343" alt="Bildschirmfoto 2021-10-18 um 13 17 57" src="https://user-images.githubusercontent.com/1142911/137721003-5ed791b2-9ebf-47f3-969e-aa70a49d9c33.png">

**Reply by shogunpurple** [reply]
Do you get any errors in the browser console?

**Reply by PatrickHuetter** [reply]
Yes there are some 400 and 403 errors.
<img width="1627" alt="Bildschirmfoto 2021-10-18 um 13 31 26" src="https://user-images.githubusercontent.com/1142911/137722707-48fd3d44-c69a-49ea-aedc-7b46de33f81e.png">

**Reply by shogunpurple** [reply]
Ok thanks for that! 

Could I see the response of the 400 call to `/init`? That should shed some light on the issue.

**Reply by PatrickHuetter** [reply]
Okay damn, seems that it's reusing my old PV. 
```
{"message":"Email address user@somedomain.tld already in use.","status":400}
```

So i will delete the PV and reinstall. It would be nice if you could give some error message on that in the frontend.

**Reply by shogunpurple** [reply]
Thanks @PatrickHuetter - yeah that's a good suggestion, right now that error message is a bit generic. We should display the message from the error instead. I've updated this on my branch, so it will be in the next budibase release.

**Reply by PatrickHuetter** [reply]
Okay i did it as described but also after full fresh install no possiblitity to create the admin user.
Now i get the following in the first request to init:
`{"message":"Connection is closed.","status":400}`

**Reply by PatrickHuetter** [reply]
@shogunpurple If you want i can share my screen and we can figure it out together. Just let us connect via Slack, Discord, Meet etc.

**Reply by zwift-mark** [reply]
Hi, 
i'm experiencing possibly a similar issue.  I'm getting a 500 error on init
```{message: "Unable to retrieve user list", status: 500}```
coudb pods show error messages like:
```
[error] 2022-10-01T00:52:26.754076Z couchdb@budibase-couchdb-2.budibase-couchdb.budibase.svc.cluster.local emulator -------- Error in process <0.4942.1> on node 'couchdb@budibase-couchdb-2.budibase-couchdb.budibase.svc.cluster.local' with exit value:                                 │
│ {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,400}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,375}]},{mem3_shards,load_shards_from_disk,2,[{file,"src/mem3_shards.erl"},{line,404}]},{mem3_shard │
│ s,for_docid,3,[{file,"src/mem3_shards.erl"},{line,97}]},{fabric_doc_open,go,3,[{file,"src/fabric_doc_open.erl"},{line,39}]},{chttpd_auth_cache,ensure_auth_ddoc_exists,2,[{file,"src/chttpd_auth_cache.erl"},{line,198}]},{chttpd_auth_cache,listen_for_changes,1,[{file,"src/chttpd_auth_ │
│ cache.erl"},{line,145}]}]}
```
(k8s AWS cluster using budibase helm chart version 0.2.11)

thoughts?

**Reply by mpkayeuk** [reply]
> Hi, i'm experiencing possibly a similar issue. I'm getting a 500 error on init `{message: "Unable to retrieve user list", status: 500}` coudb pods show error messages like:
> 
> ```
> [error] 2022-10-01T00:52:26.754076Z couchdb@budibase-couchdb-2.budibase-couchdb.budibase.svc.cluster.local emulator -------- Error in process <0.4942.1> on node 'couchdb@budibase-couchdb-2.budibase-couchdb.budibase.svc.cluster.local' with exit value:                                 │
> │ {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,400}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,375}]},{mem3_shards,load_shards_from_disk,2,[{file,"src/mem3_shards.erl"},{line,404}]},{mem3_shard │
> │ s,for_docid,3,[{file,"src/mem3_shards.erl"},{line,97}]},{fabric_doc_open,go,3,[{file,"src/fabric_doc_open.erl"},{line,39}]},{chttpd_auth_cache,ensure_auth_ddoc_exists,2,[{file,"src/chttpd_auth_cache.erl"},{line,198}]},{chttpd_auth_cache,listen_for_changes,1,[{file,"src/chttpd_auth_ │
> │ cache.erl"},{line,145}]}]}
> ```
> 
> (k8s AWS cluster using budibase helm chart version 0.2.11)
> 
> thoughts?

Exactly the same situation for me, but running on a local Docker Desktop single node Kubernetes cluster.

**Comment by ht512512** [comment]
I am running into this issue with local installation with docker-compose. Tried installation with and without BB_ADMIN_USER_EMAIL and BB_ADMIN_USER_PASSWORD defined in .env. With the variables set it did not create the user and without the variables it did not show Create Root .... screen; just login screen in both cases.
